### ISSUE TITLE
🐛 biome processing .vscode/*.txt that it shouldn't

### ISSUE THREAD

**Issue by unional** [biomejs/biome#4391]
### Environment information

```block
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.18.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.12.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

It is scanning and processing `.txt` file it shouldn't.

```sh
pnpm i
pnpm check
```

### Expected result

no error

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
What kind of error does Biome output?

**Comment by unional** [comment]
Things like string should be in qoutes. 
Seems like biome is treating them like js/ts/json file

**Comment by ematipico** [comment]
Is it possible that your editor is mapping those files to another language? 

Is it possible that there's an extension that does that? 

To make sure it's Biome and not something in your editor, always use the CLI.

**Comment by unional** [comment]
It is cli, not ide

**Comment by ematipico** [comment]
It seems those files are parsed as JSON files

**Comment by ematipico** [comment]
@Conaclos could you please a look at this issue? 

I found the bug, the function that pull the known JSON files is incorrectly assuming that all files inside the known folders are JSON files:

```rs
        if let Some(Component::Normal(parent_dir)) = path.components().rev().nth(1) {
            if Self::is_well_known_json_allow_comments_directory(parent_dir) {
                return Ok(Self::json_allow_comments());
            }
        }
```

This is incorrect, it seems, and we should still look at the extension of the file.
